lib/pbench/cli/server/repair.py

```diff
@@ -0,0 +1,320 @@
+        if not path:
+            path_repairs += 1
+            path = find_tarball(dataset.resource_id, kwargs)
+            if path:
+                detailer.message(
+                    f"{dataset.name} has no {Metadata.TARBALL_PATH}: setting {path}"
+                )
+                fix["metadata"][Metadata.TARBALL_PATH] = str(path)
+            else:
+                path_repairs_failed += 1
+                detailer.error(f"{dataset.name} doesn't seem to have a tarball")
+        if not metadata or not metadata.value:
+            metalog_repairs += 1
+            which = "metadata.log"
+            try:
+                metalog = cache_manager.Tarball._get_metadata(path)
+            except Exception:
+                which = "default"
+                metalog = {
+                    "pbench": {
+                        "name": dataset.name,
+                        "script": Metadata.SERVER_BENCHMARK_UNKNOWN,
+                    }
+                }
+                fix["metadata"][Metadata.SERVER_ARCHIVE] = True
+            detailer.message(
+                f"{dataset.name} has no {Metadata.METALOG}: setting from {which}"
+            )
+            fix["metalog"] = metalog
+        else:
+            metalog = metadata.value
+
+        if not expiration:
+            expiration_repairs += 1
+            try:
+                retention_days = get_retention_days(kwargs.get("_config"))
+                retention = datetime.timedelta(days=retention_days)
+                deletion = UtcTimeHelper(dataset.uploaded + retention).to_iso_string()
+                fix["metadata"][Metadata.SERVER_DELETION] = deletion
+                detailer.message(
+                    f"{dataset.name} {Metadata.SERVER_DELETION} "
+                    f"set ({retention_days} days) to {deletion}"
+                )
+            except Exception as e:
+                detailer.error(
+                    f"unable to calculate {dataset.name} expiration: {str(e)!r}"
+                )
+                expiration_repairs_failed += 1
+
+        if not benchmark:
+            benchmark_repairs += 1
+            script = metalog.get("pbench", {}).get(
+                "script", Metadata.SERVER_BENCHMARK_UNKNOWN
+            )
+            detailer.message(
+                f"{dataset.name} has no {Metadata.SERVER_BENCHMARK}: setting {script!r}"
+            )
+            fix["metadata"][Metadata.SERVER_BENCHMARK] = script
+        defer.append(fix)
+    for each in defer:
+        dataset = each["dataset"]
+        metalog = each.get("metalog")
+        if metalog:
+            try:
+                Metadata.create(dataset=dataset, key=Metadata.METALOG, value=metalog)
+            except Exception as e:
+                metalog_repairs_failed += 1
+                detailer.error(f"Unable to create {dataset.name} metalog: {str(e)!r}")
+        for key, value in each.get("metadata", {}).items():
+            Metadata.setvalue(dataset, key, value)
```

Comment:
We did an explicit database-commit at line 73...no analogous operation is necessary/appropriate here?

Reply:
`setvalue` and `getvalue` are higher level operations around the DB operations; the update and commit in `setvalue` are implicit.